homeassistant/components/opnsense/config_flow.py

```diff
@@ -0,0 +1,130 @@
+"""Config flow for Opnsense integration."""
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_NAME,
+    CONF_SCAN_INTERVAL,
+    CONF_TIMEOUT,
+    CONF_URL,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.selector import (
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+)
+
+from .const import (
+    CONF_API_SECRET,
+    CONF_TRACKER_INTERFACE,
+    DEFAULT_NAME,
+    DEFAULT_SCAN_INTERVAL,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+)
+
+
+class OPNSenseConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """OPNSense config flow."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle manual initiation of the config flow."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=vol.Schema(
+                    {
+                        vol.Required(CONF_NAME, default=DEFAULT_NAME): str,
+                        vol.Required(CONF_URL): str,
+                        vol.Required(CONF_API_KEY): str,
+                        vol.Required(CONF_API_SECRET): str,
+                        vol.Optional(CONF_VERIFY_SSL, default=False): bool,
+                        vol.Optional(
+                            CONF_TRACKER_INTERFACE, default=[]
+                        ): SelectSelector(
+                            SelectSelectorConfig(
+                                options=[],
+                                multiple=True,
+                                custom_value=True,
+                                mode=SelectSelectorMode.DROPDOWN,
+                            )
+                        ),
+                    }
+                ),
+            )
+
+        self._async_abort_entries_match({CONF_NAME: user_input[CONF_NAME]})
+        return self.async_create_entry(
+            title=user_input[CONF_NAME],
+            data={},
+            options={**user_input},
+        )
+
+    async def async_step_import(self, import_info: dict[str, Any]) -> FlowResult:
+        """Import OPNSense config from configuration.yaml."""
+        return await self.async_step_user(user_input=import_info)
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> config_entries.OptionsFlow:
+        """Create the options flow."""
+        return OptionsFlowHandler(config_entry)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
```

Comment:
Please leave the options flow for a follow up PR

Reply:
Any specific reason for that? Just seems like more work reviewing 2 PRs that could be one?

Reply:
To cut down this PR in size. The less content there is, the less the PR can get stuck on. Also, this is a change which isn't hard coupled with the rest, so it's easy to add and easy to remove